### ISSUE TITLE
Divergence rick & sean repos

### ISSUE THREAD

**Issue by rgleason** [seandepagnier/watchdog_pi#123]
Sean, I think I have identified the reasons these have diverge.

1. From SD repos - signalk_client changed to pypilot_client.h with changes.
1. From RG repos - Rudder Angle added.
2. From RG repos - Watchdog.h moved to include dir
3. There are some other changes to chase down, like the date for Formbuilder etc.

But I think I can fix it.

**Comment by rgleason** [comment]
We need the beta version tested, particularly the Pypilot Alarms.

We need the beta version of watchdog tested.

https://www.cruisersforum.com/forums/f134/watchdog-plugin-150313.html#post3674963

Watchdog 2.4.45.2 [master-chg2sean](https://github.com/rgleason/watchdog_pi/tree/master-chg2sean) -fix divergence in branch and add Pypilot.
Volunteer Testers,
We need to have watchdog_pi v2.4.45.2 Tested, particularly for the the new pypilot functions from Sean's repository.
Watchdog_pi v2.4.45.2 will be found in the PIM Beta directory once bdbcat merges my pull request and you update your Beta Catalog. Please post any issues in the [Watchdog Thread](https://www.cruisersforum.com/forums/f134/watchdog-plugin-150313.html) or in the Test spreadsheet.

**Comment by rgleason** [comment]
**Watchdog Diverged - Rgleason and Sean - Fix**

What we have to do is take our current master, duplicate it and then modify it with "new" history by

reverting some commits and then cherry picking some commits from sean-master and committing.

-------------------------------------------------------------------

**-This will make an identical copy of local master**

    git checkout -b rg-master origin/master

**-Now make several patches of the Rudder changes  f8d93a2 and 7d0dc50  for later referral and use.**

    git format-patch -1 <commit_sha>  -o patches

    Then copy these patches from patch directory to somewhere safe in source.

    git format-patch -1 f8d93a2  -o patches

    git format-patch -1 7d0dc50  -o patches

**-Now revert the two rudder commits because these commits will screw up a clean merge of Sean's changes.**

Revert is safer because it does not change history, and there is a specific commit in history which reverts it.

f8d93a2 Merge pull request #38 from pa2wlt/with_rudder
7d0dc50 Added RudderAlarm and set VERSION_TWEAK to 1.

    git revert f8d93a2 -m 1
    git add
    git commit -am "revert pa2wlt/with_rudder"

    git revert 7d0dc50
    git add
    git commit -am "revert Add RudderAlarm set vers tweak"

and then push to remote.

**-This will get an identical copy of upstream/master and make it as a new branch named sean-master**

    git checkout -b sean-master upstream/master

**- Then to make rg-master  similar to Sean's master, from rg-master,  we need to cherry pick missing commits from sean-master.**
Like git merge, the git cherry-pick command is one you use to incorporate changes from other branches into your current branch. How do they differ, though?
When merging, you’ll typically want to incorporate all the changes made in the other branch into the current one.
But sometimes you want only some of the changes. In these cases, cherry-picking comes in handy.
A classical use case for git cherry-pick would be to backport a bug fix to an older version of the application.

For conflicts cancel the cherry-pick operation by running git cherry-pick --abort.

From the branch you want to the cherry pick to land in branch  rg-master

        git cherry-pick -x <sha>  <sha> <sha> <sha>

        git cherry-pick -x <76f8496>  <95c1c1606> <6b5293f5e> <39731d23> <b7dd9036c> <0e982d01ab>

        This could be made simpler with

        git cherry-pick -x <sha>^..<sha>

        git cherry-pick -x 76f8496^ ..b7dd903    This fails due to some merge I could not identify.

        I had to do it separately, like this, and then fix the conflicts each time.

        git cherry-pick -x 76f8496

    1/6/22   add wddc    Don't have 76f8496      DONE
    https://github.com/seandepagnier/watchdog_pi/commit/76f849627c4831ab5a4bfc5dc274d21b48797b9f

    1/6/22     add close      Don't have 95c1c1606    DONE
    https://github.com/seandepagnier/watchdog_pi/commit/95c1c16062532b8eb5b04a199f364b4b493b4465

    1/6/22    improve pypilot watchdog alarm         Don't have 6b5293f5e  DONE
    https://github.com/seandepagnier/watchdog_pi/commit/6b5293f5ea4d859d06028d8e8ea5f04cd852e336

    1/6/22    fix issues with pypilot alarm      Don't have 39731d23  DONE
    https://github.com/seandepagnier/watchdog_pi/commit/39731d23cdf158593f04b7b8a11834a31b97e2f5

    1/11/22  make src include  Don't have b7dd9036c   DONE
    https://github.com/seandepagnier/watchdog_pi/commit/b7dd9036ce128974ee348a429847da17bcb56848

    2/8/22   Merge branch master rgleason           2/8/22  (2/8/21) 2.4.39.0 + TP 1.0.185.0 fix macos   (only FE2)
    Already have this.   This is just updating the FE2 TP files.   Outdated now, so don't use it.

    2/8/22   fix include location    Don't have 0e982d01ab  DONE
    https://github.com/seandepagnier/watchdog_pi/commit/0e982d01abfd149924b56c8a2c7d36cf380b5e22
    This added odapi.api to includes in cmakelists.txt which may not be right.

Then see if it builds.

Then push to Sean's repos

===============================

I screwed up on 2/8/22 by not making a PR and merging Sean's changes....
What is the best way to make these 5 changes now....

1.  Make patches of my branch, back to 2/8/22   (16 patches)
2.  Don't apply the two pa2wlt  two rudder changes, but do copy the files. They have conflicts.
       Watchdog.fbp
       include/Alarm.h
       include/WatchdogUI.h
       src/Alarm.cpp
       src/NewAlarmDialog.cpp
       WatchdogUI.cpp
       CMmakeLists.txt

3.  Carefully apply the 14 patches.
3.  Carefully add the two pa2wlt rudder changes.
 


Sean                                            Rick
12/8/22  2.4.39.0 + TP 1.0.185.0 fix macos      12/8/21  2.4.39.0 + TP 1.0.185.0 fix macos
1/6/22   add wddc                               Don't have 76f8496    https://github.com/seandepagnier/watchdog_pi/commit/76f849627c4831ab5a4bfc5dc274d21b48797b9f
         add close                                      Don't have 95c1c1606  https://github.com/seandepagnier/watchdog_pi/commit/95c1c16062532b8eb5b04a199f364b4b493b4465
         improve pypilot watchdog alarm         Don't have 6b5293f5e  https://github.com/seandepagnier/watchdog_pi/commit/6b5293f5ea4d859d06028d8e8ea5f04cd852e336
         fix issues with pypilot alarm          Don't have 39731d23   https://github.com/seandepagnier/watchdog_pi/commit/39731d23cdf158593f04b7b8a11834a31b97e2f5
1/11/22  make src include                       Don't have b7dd9036c  https://github.com/seandepagnier/watchdog_pi/commit/b7dd9036ce128974ee348a429847da17bcb56848
2/8/22   Merge branch master rgleason           2/8/22  (2/8/21) 2.4.39.0 + TP 1.0.185.0 fix macos   (only FE2)
2/8/22   fix include location 0e982d01ab
       https://github.com/seandepagnier/watchdog_pi/commit/0e982d01abfd149924b56c8a2c7d36cf380b5e22

branch sean identical  how?  force to upstream?
then cherry pick from master to sean without Rudder Alarm.

                                                2/25/22  2.4.39.1 + TP 1.0.193   (only FE2)
                                                3/10/22  rm manual version no    (only manual)
                                                3/11/22  2.4.40 + TP 1.0.194     (only FE2)
                                                4/2/22     2.4.41.0 + flatpak arm64 wx3     (only FE2)                                    
                                                4/8/22   fix antora yml          (only manual)  
                                                4/12/22  Added RudderAlarm and set VERSION_TWEAK to 1.    ??
                                                4/13/22  Merge pull request #38 from pa2wlt/with_rudder   ??
                                                4/13/22  2.4.42.0                (only FE2)
                                                6/16/22  2.4.43.0 + TP 1.0.212.0  (only FE2)
                                                6/17/22  2.4.44 +TP 1.0.213 fix OS  (only FE2)
                                                7/20/22  fix manual odraw ocpn_draw link  (only manual)
                                                7/24/22  Updated pot-file           (language)
                                                8/23/22  New po-files from Crowdin   (language)
                                                8/23/22  Delete po file with error since no translaions done  (language)
                                                8/23/22  Merge pull request #39 from Hakansv/po_update crowdin translations  (langguage)
                                                8/23/22  2.4.45.0 + language  (language and version)
                                               
Added RudderAlarm and set VERSION_TWEAK to 1.  7d0dc505
https://github.com/rgleason/watchdog_pi/commit/7d0dc505596aa7039f692ec68be850ab1c3704f6

            Has changes to
            Watchdog.fbp
            Include
               Alarm.h
               WatchdogUI.h
            src
               Alarm.cpp
               NewAlarmDialog.cpp
               WatchdogUI.cpp

Merge pull request #38 from pa2wlt/with_rudder  f8d93a21
https://github.com/rgleason/watchdog_pi/commit/f8d93a210041e82b2d1cf368ae625677823e9cbe

Same as above....   

========================

Current rg-master   after revert and cherry-pick

=======================

fcgle@Hesper MINGW64 ~/source/watchdog_pi (rg-master)
$ git log --pretty=oneline --abbrev-commit
c41ff4b (HEAD -> rg-master) fix include location
323f286 make src include
add4fd7 fix issues with pypilot alarm
7a213e4 improve pypilot watchdog plugin
60664f0 add close
13d8769 add wddc
8093f3a (origin/rg-master) Revert "Merge pull request #38 from pa2wlt/with_rudder"
ed8416f (tag: v2.4.45, origin/master, origin/HEAD, master) 2.4.45.0 + language
1cd7c66 Merge pull request #39 from Hakansv/po_update
3a38c9a Delete po file with error since no translations done
fffd81c New po-files from Crowdin
753fe49 Updated pot-file
8f6c7cb fix manual odraw ocpn_draw link
5fd6210 (tag: v2.4.44) 2.4.44 +TP 1.0.213 fix OS
0f349eb (tag: v2.4.43.0) 2.4.43.0 + TP 1.0.212.0
05d871b (tag: v2.4.42) 2.4.42.0
f8d93a2 Merge pull request #38 from pa2wlt/with_rudder
7d0dc50 Added RudderAlarm and set VERSION_TWEAK to 1.
958b8ba fix antora yml
881f8e8 (tag: v2.4.41) 2.4.41.0 + flatpak arm64 wx3
d0f19af (tag: v2.4.40) 2.4.40 + TP 1.0.194
9427b58 rm manual version no
9f8c533 (tag: v2.4.39.1) 2.4.39.1 + TP 1.0.193
bf9415a (tag: v2.4.39.0) 2.4.39.0 + TP 1.0.185.0 fix macos
39fca8b (tag: v2.4.38.0) 2.4.38.0 + fix circleci config
e57f635 (tag: v2.4.37.0) 2.4.37.0 + TP 1.0.180 + imprv svg icons
25f5867 2.4.36.1 improv icon svg
ce0bc50 2.4.36.0 + TP 1.0.176.1 -correct
0f5892a (tag: v2.4.36.0) 2.4.36.0 + TP 1.0.147.1 fix android bld
70fdb07 test android 2.4.35.1
08caf3f (tag: v2.4.35.0) v2.4.35.0 + tp1.0176.1

==============================

Current sean-master (as it stands now

=============================

fcgle@Hesper MINGW64 ~/source/watchdog_pi (sean-master)
$ git log --pretty=oneline --abbrev-commit

0e982d0 (HEAD -> sean-master, upstream/master) fix include location

1c92c4d Merge branch 'master' of https://github.com/rgleason/watchdog_pi

b7dd903 make src include

39731d2 fix issues with pypilot alarm

6b5293f improve pypilot watchdog plugin -

95c1c16 add close   -Done

76f8496 add wddc   - Done

bf9415a (tag: v2.4.39.0) 2.4.39.0 + TP 1.0.185.0 fix macos
39fca8b (tag: v2.4.38.0) 2.4.38.0 + fix circleci config
e57f635 (tag: v2.4.37.0) 2.4.37.0 + TP 1.0.180 + imprv svg icons
25f5867 2.4.36.1 improv icon svg
ce0bc50 2.4.36.0 + TP 1.0.176.1 -correct
0f5892a (tag: v2.4.36.0) 2.4.36.0 + TP 1.0.147.1 fix android bld
70fdb07 test android 2.4.35.1
08caf3f (tag: v2.4.35.0) v2.4.35.0 + tp1.0176.1
b6263af v2.4.34.1 + tp1.0.176
5f60507 (tag: v2.4.34.0) v2.4.34.0 + tp1.0.175.4
711534c chng antora for nav adoc
16374cd Update index.adoc   more small changes
7392c4c Update index.adoc
b3b3ce5 Merge branch 'master' of github.com:rgleason/watchdog_pi
8d846aa manual chng link to ocpn_draw
17027af Fix Link
5532ffb Fix two links
95d3142 fix site.yml title elimin word manual
1306a7d manual sml change
60db50f manual more improvements
3822927 manual alarms outline chngs
ff65417 Merge branch 'master' of github.com:rgleason/watchdog_pi
b9435e3 manual stnd config
0ae84a1 manual wd.adoc chngs
bd928ad Merge pull request #31 from Rasbats/patch-1
42780e7 Update watchdog.adoc
831c744 Merge pull request #30 from leamas/rick-pr2
cf48b57 Replace dokuwiki reference to Draw Messainng page.
29782fe merge changes..  no idea if it works Merge branch 'master' of https://github.com/rgleason/watchdog_pi into rgleason-master
7f7c655 update pypilot support
e17ecfb Merge pull request #29 from leamas/fix-some-links
e22fe9b Fix some dokuwiki links (one left)
19850bc fix more links + remove images

========================

LOG when rg-master started.

=============================
fcgle@Hesper MINGW64 ~/source/watchdog_pi (master)
$ git log --pretty=oneline --abbrev-commit
ed8416f (HEAD -> master, tag: v2.4.45, origin/master, origin/HEAD) 2.4.45.0 + language
1cd7c66 Merge pull request #39 from Hakansv/po_update
3a38c9a Delete po file with error since no translations done
fffd81c New po-files from Crowdin
753fe49 Updated pot-file
8f6c7cb fix manual odraw ocpn_draw link
5fd6210 (tag: v2.4.44) 2.4.44 +TP 1.0.213 fix OS
0f349eb (tag: v2.4.43.0) 2.4.43.0 + TP 1.0.212.0
05d871b (tag: v2.4.42) 2.4.42.0
--
f8d93a2 Merge pull request #38 from pa2wlt/with_rudder
7d0dc50 Added RudderAlarm and set VERSION_TWEAK to 1.
--
958b8ba fix antora yml
881f8e8 (tag: v2.4.41) 2.4.41.0 + flatpak arm64 wx3
d0f19af (tag: v2.4.40) 2.4.40 + TP 1.0.194
9427b58 rm manual version no
9f8c533 (tag: v2.4.39.1) 2.4.39.1 + TP 1.0.193
--
would like to insert Sean's commits here, but that is not allowed as it changes history.
so I will add them at the end when I've removed the rudder changes.

--
bf9415a (tag: v2.4.39.0) 2.4.39.0 + TP 1.0.185.0 fix macos
39fca8b (tag: v2.4.38.0) 2.4.38.0 + fix circleci config
e57f635 (tag: v2.4.37.0) 2.4.37.0 + TP 1.0.180 + imprv svg icons
25f5867 2.4.36.1 improv icon svg
ce0bc50 2.4.36.0 + TP 1.0.176.1 -correct
0f5892a (tag: v2.4.36.0) 2.4.36.0 + TP 1.0.147.1 fix android bld
:

**Comment by rgleason** [comment]
Note that 
https://github.com/seandepagnier/watchdog_pi/commit/8093f3a6d3592d60f49b561572805c7744343d8d (origin/rg-master) Revert "Merge pull request https://github.com/seandepagnier/watchdog_pi/pull/38 from pa2wlt/with_rudder"

will need to be redone with a new PR.

**Comment by rgleason** [comment]
Errors when building for Win  with wxWidgets 3.1.2 for OpenCPN 5.6.2

```
C:\project\ocpn_project\src\watchdog_pi.cpp(111): error C2065: 'm_pypilot_host': undeclared identifier [C:\project\ocpn_project\build\watchdog_pi.vcxproj]
  WatchdogDialog.cpp
  WatchdogPropertiesDialog.cpp
  ConfigurationDialog.cpp
  NewAlarmDialog.cpp
  EditAlarmDialog.cpp
  Alarm.cpp
C:\project\ocpn_project\src\Alarm.cpp(113): error C2248: 'DepthPanel::m_cUnits': cannot access protected member declared in class 'DepthPanel' [C:\project\ocpn_project\build\watchdog_pi.vcxproj]
  c:\project\ocpn_project\src\WatchdogUI.h(421): note: see declaration of 'DepthPanel::m_cUnits'
  c:\project\ocpn_project\src\WatchdogUI.h(414): note: see declaration of 'DepthPanel'
C:\project\ocpn_project\src\Alarm.cpp(2587): error C2504: 'pypilotClient': base class undefined [C:\project\ocpn_project\build\watchdog_pi.vcxproj]
C:\project\ocpn_project\src\Alarm.cpp(2590): error C2614: 'pypilotAlarm': illegal member initialization: 'SignalKClient' is not a base or member [C:\project\ocpn_project\build\watchdog_pi.vcxproj]
C:\project\ocpn_project\src\Alarm.cpp(2626): error C3861: 'connected': identifier not found [C:\project\ocpn_project\build\watchdog_pi.vcxproj]
C:\project\ocpn_project\src\Alarm.cpp(2665): error C3861: 'connected': identifier not found [C:\project\ocpn_project\build\watchdog_pi.vcxproj]
C:\project\ocpn_project\src\Alarm.cpp(2666): error C2039: 'm_pypilot_host': is not a member of 'watchdog_pi' [C:\project\ocpn_project\build\watchdog_pi.vcxproj]
  C:\project\ocpn_project\include\WatchdogDialog.h(48): note: see declaration of 'watchdog_pi'
C:\project\ocpn_project\src\Alarm.cpp(2667): error C2039: 'm_pypilot_host': is not a member of 'watchdog_pi' [C:\project\ocpn_project\build\watchdog_pi.vcxproj]
  C:\project\ocpn_project\include\WatchdogDialog.h(48): note: see declaration of 'watchdog_pi'
C:\project\ocpn_project\src\Alarm.cpp(2672): error C2660: 'connect': function does not take 1 arguments [C:\project\ocpn_project\build\watchdog_pi.vcxproj]
  C:\Program Files (x86)\Microsoft SDKs\Windows\v7.1A\include\winsock.h(754): note: see declaration of 'connect' (compiling source file C:\project\ocpn_project\src\Alarm.cpp)
C:\project\ocpn_project\src\Alarm.cpp(2680): error C3861: 'receive': identifier not found [C:\project\ocpn_project\build\watchdog_pi.vcxproj]
C:\project\ocpn_project\src\Alarm.cpp(2688): error C3861: 'connected': identifier not found [C:\project\ocpn_project\build\watchdog_pi.vcxproj]
C:\project\ocpn_project\src\Alarm.cpp(2705): error C3861: 'update_watchlist': identifier not found [C:\project\ocpn_project\build\watchdog_pi.vcxproj]
C:\project\ocpn_project\src\Alarm.cpp(2733): error C2039: 'm_cHost': is not a member of 'pypilotPanel' [C:\project\ocpn_project\build\watchdog_pi.vcxproj]
  c:\project\ocpn_project\src\WatchdogUI.h(502): note: see declaration of 'pypilotPanel'
C:\project\ocpn_project\src\Alarm.cpp(2755): error C2039: 'm_cHost': is not a member of 'pypilotPanel' [C:\project\ocpn_project\build\watchdog_pi.vcxproj]
  c:\project\ocpn_project\src\WatchdogUI.h(502): note: see declaration of 'pypilotPanel'
  pypilot_client.cpp
```

**Comment by rgleason** [comment]
bdbcat and I fixed this.  Fixed by 2.4.70
